HiP-DataStore/Controllers/MediaController.cs

```diff
@@ -78,16 +79,22 @@ public IActionResult Get(MediaQueryArgs args)
                     .FilterByStatus(args.Status)
                     .FilterIf(args.Used != null, x => x.IsUsed == args.Used)
                     .FilterIf(args.Type != null, x => x.Type == args.Type)
-                    .FilterIf(args.Timestamp != null, x => DateTimeOffset.Compare(x.Timestamp, args.Timestamp.GetValueOrDefault()) == 1)
-                    .FilterIf(!string.IsNullOrEmpty(args.Query), x =>
+                     .FilterIf(!string.IsNullOrEmpty(args.Query), x =>
                         x.Title.ToLower().Contains(args.Query.ToLower()) ||
                         x.Description.ToLower().Contains(args.Query.ToLower()))
                     .Sort(args.OrderBy,
                         ("id", x => x.Id),
                         ("title", x => x.Title),
                         ("timestamp", x => x.Timestamp))
                     .Paginate(args.Page, args.PageSize)
-                    .Select(x => new MediaResult
+                    .ToList();
+
+                medias = medias.AsQueryable().FilterIf(args.Timestamp != null, x => DateTimeOffset.Compare(x.Timestamp, args.Timestamp.GetValueOrDefault()) == 1).ToList();
```

Comment:
Hm, I think filtering after pagination is dangerous... Imagine at line 85 there are 10 items and a page size of 5, then line 89 would return the first 5 items (i.e. the first page). Now, filtering by timestamp is done AFTER pagination, so line 92 might throw out some of these 5 items. Thus, in the end you might end up with page 1 of 2 having less items than the desired page size.